kzg/src/eth/eip_7594.rs

```diff
@@ -0,0 +1,146 @@
+    blob: [u8; BYTES_PER_BLOB],
+    das: &impl DAS<B>,
+) -> Result<CellsKzgProofs, String>
+where
+    B::G1: Copy,
+    B::Fr: Copy,
+{
+    let blob = bytes_to_blob(&blob)?;
+
+    let mut recovered_cells = [B::Fr::default(); FIELD_ELEMENTS_PER_EXT_BLOB];
+    let mut recovered_proofs = [B::G1::default(); CELLS_PER_EXT_BLOB];
+
+    let _ = das.compute_cells_and_kzg_proofs(
+        Some(&mut recovered_cells),
+        Some(&mut recovered_proofs),
+        &blob,
+    );
+
+    let converted_cells = cells_elements_to_cells_bytes::<B>(&recovered_cells)?;
+    let converted_proofs = recovered_proofs
+        .into_iter()
+        .map(|proof| proof.to_bytes())
+        .collect::<Vec<_>>();
+
+    Ok((converted_cells, converted_proofs))
+}
+
+pub fn verify_cell_kzg_proof_batch_raw<B: EcBackend>(
+    commitments: &[[u8; BYTES_PER_COMMITMENT]],
+    cell_indices: &[usize],
+    cells: &[[u8; BYTES_PER_CELL]],
+    proofs: &[[u8; BYTES_PER_PROOF]],
+    das: &impl DAS<B>,
+) -> Result<bool, String> {
+    let commitments = cfg_iter!(commitments)
+        .map(|commitment| B::G1::from_bytes(commitment))
+        .collect::<Result<Vec<_>, _>>()?;
+
+    let nested_cells = cfg_iter!(cells)
+        .map(|bytes| bytes_to_cell::<B>(bytes))
+        .collect::<Result<Vec<Vec<_>>, _>>()?;
+    let cells = nested_cells.into_iter().flatten().collect::<Vec<_>>();
+
+    let proofs = cfg_iter!(proofs)
+        .map(|proof| B::G1::from_bytes(proof))
+        .collect::<Result<Vec<_>, _>>()?;
+
+    das.verify_cell_kzg_proof_batch(&commitments, cell_indices, &cells, &proofs)
+}
+
+fn bytes_to_cell<B: EcBackend>(bytes: &[u8]) -> Result<Vec<B::Fr>, String> {
+    if bytes.len() != BYTES_PER_CELL {
+        return Err(format!(
+            "Invalid byte length. Expected {} got {}",
+            BYTES_PER_CELL,
+            bytes.len(),
+        ));
+    }
+
+    bytes
+        .chunks(BYTES_PER_FIELD_ELEMENT)
+        .map(B::Fr::from_bytes)
+        .collect()
+}
+
+fn cells_elements_to_cells_bytes<B: EcBackend>(
+    bytes: &[B::Fr],
+) -> Result<Vec<[u8; BYTES_PER_CELL]>, String> {
+    // NOTE: chunk_size = BYTES_PER_CELL / BYTES_PER_FIELD_ELEMENT
+    let chunk_size = FIELD_ELEMENTS_PER_CELL;
+    if bytes.len() % chunk_size != 0 {
+        return Err(format!("Invalid byte length. got {}", bytes.len(),));
+    }
+
+    bytes
+        .chunks(chunk_size)
+        .map(|cell_bytes| {
+            let mut result = [0u8; BYTES_PER_CELL];
+            for (idx, field_element) in cell_bytes.iter().enumerate() {
+                let bytes_element = field_element.to_bytes();
+                let start = idx * BYTES_PER_FIELD_ELEMENT;
+                let end = start + BYTES_PER_FIELD_ELEMENT;
+                if end > BYTES_PER_CELL {
+                    return Err(format!(
+                        "Invalid cell byte length. Expected {} got {}",
+                        BYTES_PER_CELL, end,
+                    ));
+                }
```

Comment:
Looks like this check is unnecessary, as `chunks` function already returns slices, with size <= chunk_size.